src/getRecords.ts

```diff
@@ -0,0 +1,57 @@
+// IDEA: getRecords handles cases of:
+// - getting all records from a table
+// - getting specific records from a table (single or multiple)
+// the difference is handled by either supplying a set of record id's or not
+// (and other params like filterByFormula etc)
+
+// also - can either call this once the airtable singleton has been set up (in which case "airtable" is passed to the function under the hood), or call it by itself and pass the api key explicitly
+
+import AirtableSDK from "airtable"
+
+import { RecordZ } from "./types/record"
+import { SchemaZodT } from "./types/schema"
+import { SelectQueryParamsT } from "./types/queryParams"
+
+export interface GetRecords<T extends SchemaZodT> {
+	tableId: string
+	baseId: string
+	schema: T
+	query?: SelectQueryParamsT
+}
```

Comment:
Yeah that's a great call - I definitely default away from zod functions because I don't find them that useful in app code (not often that a function needs runtime parsing of arguments in app?). But I get the point that it makes a lot of sense here and in this kind of situation in general :)!

Reply:
> not often that a function needs runtime parsing of arguments in app

For generalised pure utility functions I think it is ideal to parse the args as the likelihood is that the function wouldn't work anyway with the wrong args being supplied. Most likely this happens during development and testing and so when it blows up on a failure it tells the developer there is something wrong with their logic before they call the function.

Hence I think it makes a lot of sense for a library of utility functions to use this approach.